FreeRTOS_DHCP.c

```diff
@@ -1449,8 +1451,10 @@
                  * returned to the stack. */
                 FreeRTOS_ReleaseUDPPayloadBuffer( pucUDPPayloadBuffer );
             }
-
-            xResult = pdTRUE;
+            else
+            {
+                xResult = pdPASS;
+            }
```

Comment:
You are right, it makes no sense to wait for a reply to a messages that was never sent.